kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/ConsumerFactoryImpl.java

```diff
@@ -19,50 +19,22 @@
 package io.tabular.iceberg.connect.channel;
 
 import java.util.Map;
-import java.util.UUID;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
-import org.apache.kafka.clients.admin.Admin;
 import org.apache.kafka.clients.consumer.Consumer;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
-import org.apache.kafka.clients.producer.KafkaProducer;
-import org.apache.kafka.clients.producer.Producer;
-import org.apache.kafka.clients.producer.ProducerConfig;
 import org.apache.kafka.common.serialization.ByteArrayDeserializer;
-import org.apache.kafka.common.serialization.ByteArraySerializer;
 import org.apache.kafka.common.serialization.StringDeserializer;
-import org.apache.kafka.common.serialization.StringSerializer;
 
-public class KafkaClientFactory {
-  private final Map<String, String> kafkaProps;
-
-  public KafkaClientFactory(Map<String, String> kafkaProps) {
-    this.kafkaProps = kafkaProps;
-  }
-
-  public Producer<String, byte[]> createProducer(String transactionalId) {
-    Map<String, Object> producerProps = Maps.newHashMap(kafkaProps);
-    producerProps.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId);
-    producerProps.put(ProducerConfig.CLIENT_ID_CONFIG, UUID.randomUUID().toString());
-    KafkaProducer<String, byte[]> result =
-        new KafkaProducer<>(producerProps, new StringSerializer(), new ByteArraySerializer());
-    result.initTransactions();
-    return result;
-  }
-
-  public Consumer<String, byte[]> createConsumer(String consumerGroupId) {
-    Map<String, Object> consumerProps = Maps.newHashMap(kafkaProps);
+public class ConsumerFactoryImpl implements ConsumerFactory {
+  @Override
+  public Consumer<String, byte[]> create(Map<String, String> props) {
+    Map<String, Object> consumerProps = Maps.newHashMap(props);
+    // TODO: why putIfAbsent? why not just put?
     consumerProps.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
```

Comment:
I'm not sure I understand why this is a `putIfAbsent` rather than just `put`
I'd like to know why. 
To me, the most correct thing here is to always start from latest.

Reply:
I'm with you, I can't think of a good reason why you would set this to earliest.  All this is doing is leaving it to be user configurable w/ a default to latest.  Can live with it if we can't figure out the mystery.